packages/vite/src/vite-node.ts

```diff
@@ -102,7 +102,7 @@ function createViteNodeMiddleware (ctx: ViteBuildContext, invalidates: Set<strin
     node.shouldExternalize = async (id: string) => {
       const result = await isExternal(id)
       if (result?.external) {
-        return resolveModule(result.id, { url: ctx.nuxt.options.rootDir })
+        return resolveModule(result.id, { url: ctx.nuxt.options.modulesDir })
```

Comment:
@antfu Shall we do this to be same as [vite-node](https://github.com/vitest-dev/vitest/blob/95da4d66933db46da368d5d66f422afd26fba4c1/packages/vite-node/src/externalize.ts#L85)?

```suggestion
  return id
```

Reply:
The id passed here should already be resolved by Vite. So yes I think we could directly do this.

Reply:
Thanks for checking. FYI, it is not always resolved. You can reproduce it only with an external repository by putting a debugger. `vue` for instance is directly passed to `shouldExternalize`.